style: use f-strings to format str

mwtoews opened the pull request:
Python 3.6 introduced f-strings ([PEP 498](https://www.python.org/dev/peps/pep-0498/)). For example, these are the types of changes made in this PR:
```diff
-        "Mean R/T is between {} and {}".format(RTmin, RTmax)
+        f"Mean R/T is between {RTmin} and {RTmax}"
```
or
```diff
-        f_wel.write("%s\n" % self.heading)
+        f_wel.write(f"{self.heading}\n")
```
Advantages of f-strings:

- Easier to read, and less error prone, as the items are directly formatted where they should
- Quicker to read and type, because they are shorter
- Similar or better performance than other styles of string formatting
- Works with all supported Python versions, so no incompatibility issues

---

This patch was quick to prepare, using [flynt](https://github.com/ikamensh/flynt), and has not been extensively reviewed. It is also currently in draft as I'm not 100% sure it's ideal. It's potentially disruptive, as nearly every file is touched (similar to #954). Feedback welcome as to if this contribution would be useful, then I can take further review.

langevin-usgs commented:
I'm a fan of f-strings.  If we can be reasonably sure that this doesn't break anything and that messages and strings will behave as before, then I'm all for it.  And with no other PRs in the queue that we know about, now is a good time.

jdhughes-usgs commented:
I am ok with the changes but I think the PR should be reviewed change by change to make sure that a fixed length format or type has not been inadvertently changed. I will spend an hour or so on it today if possible approve file by file. Hopefully we can get this PR approved in the next day or two.

@mwtoews could you address the codacy issues. I realize they are not yours but it would be good to clean these things up as we make changes.

mwtoews commented:
Ready for review. I've reviewed these edits several times already.

A few (too many) notes:

- Most work was automated with flynt, also using `--aggressive` and `--transform-concats` options.
- I've rejected automated edits that didn't improve readability, or are potentially worth future refactoring (~5% ish)
- Some old %-style fixed width formatting (e.g. mtadv.py) were kept in the old style, as there are subtle behavior differences that may cause bugs (e.g. flopy_swi2_ex2.py assigns a float `mxpart=1e7`, but this needs to be formatted as an int, so `%10i` works, but `:10d` does not, and `:10.0f` works, but looks strange.)
- Flynt does not currently work for mult-line strings, so there is still a great deal other code that was untouched.
- Manual edits to automated edits were necessary, although I tried my best to not get carried away, as manual edits are more error prone.
- Literal formats were simplified, e.g. `"{0} {1}".format(0, x)` would normally be `f"{0} {x}"`, but were edited to be `f"0 {x}"` instead.
- Removed unnecessary `:s` (string) and `:d` (decimal) presentation types, as they don't alter the string outputs.
- Bare float presentation types `:f` were kept, although keep in mind it has a default 6 digits of precision after the decimal point.
- Conversion fields were used, e.g. `f{str(e)}` -> `f{e!s}`, which is equivalent.
- The formatted string outputs should remain identical as before, with the exception of one minor fix to a filename:`f"{model.name}..mp.advobs"` -> `f"{model.name}.mp.advobs"`
- Added TODO notes for `flopy/utils/util_array.py` and `flopy/mf6/data/mffileaccess.py`.
- The codacy issues are primarily due to use of variable names less than three characters, which is beyond the scope of this PR to update/rename. (Many other tempting changes were not done to limit the scope and intent of this PR).

langevin-usgs reviewed (approved):
I looked through all of these changes. My visual checks, though not rigorous, indicate that this is pretty solid and ready to go.  Will need to keep on the lookout for any weirdness, but I'd say let's merge it in.  f-strings are more concise and straightforward.  Thanks @mwtoews.

mwtoews commented on flopy/modflow/mf.py:
```diff
@@ -968,30 +935,28 @@ def load(
-                    "   The following {} packages were not loaded.".format(
-                        len(files_not_loaded)
-                    )
+                    "   The following {len(files_not_loaded)} packages "
```
Whoops! This was one of the few manual edits, and is missing an `f`. I'll follow this up within another refactor touching this file probably in the next week or so.
